### PR TITLE
[WebNN EP] Check if the tensor shape has 0 dimension

### PR DESCRIPTION
**Pull request by shiyi9801**
WebNN doesn't support empty tensor.

### CODE REVIEW

**Comment by shiyi9801** [comment]
@Honry PTAL, thanks!

File: onnxruntime/core/providers/webnn/builders/impl/expand_op_builder.cc [hunk 1]
```diff
@@ -88,6 +88,10 @@ bool ExpandOpBuilder::IsOpSupportedImpl(const InitializedTensorSet& initializers
     LOGS(logger, VERBOSE) << "Cannot get shape.";
     return false;
   }
+  if (std::any_of(new_shape.begin(), new_shape.end(), [](int64_t dimension) { return dimension == 0; })) {
+    LOGS(logger, VERBOSE) << "Expand does not support new shape with 0 dimension.";
+    return false;
+  }
```
Comment (Honry) [inline onnxruntime/core/providers/webnn/builders/impl/expand_op_builder.cc:94]: Could you also fix the `reshape_op_builder.cc` ?

Comment (shiyi9801) [reply onnxruntime/core/providers/webnn/builders/impl/expand_op_builder.cc:94]: It's already check in `reshape_op_builder.cc`.

**Comment by shiyi9801** [comment]
@Honry @fdwr @guschmue PTAL, thanks!

**Review by Honry — APPROVED** [review]
👍

**Review by guschmue — APPROVED** [review]

**Review by fdwr — APPROVED** [review]
So we might remove this later, when/if WebNN supports 0 dimensions, but it's prudent for now.

**Pull request merged**